src/CodeGen_ARM.cpp

```diff
@@ -149,6 +149,18 @@ CodeGen_ARM::CodeGen_ARM(Target target)
         }
     }
 
+    casts.emplace_back("vqdmulh.v4i16", "sqdmulh.v4i16", 4,
```

Comment:
So this PR is really "add some cast cases we didn't handle before, plus testing for them in simd_op_check"? Probably worth updating the commit/PR title then, I had assumed the change was just to the test, not the codegen.